src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -693,6 +639,166 @@ class EraseGather : public ngraph::pass::MatcherPass {
 };
 }  // namespace fuse
 
+namespace back_prop {
+class Binary : public ngraph::pass::MatcherPass {
+public:
+    Binary() {
+        MATCHER_SCOPE(Binary);
+        auto pattern_root =
+            pattern::wrap_type<op::util::BinaryElementwiseArithmetic, opset8::FakeQuantize>(pattern::has_static_rank());
+
+        auto callback = [=](pattern::Matcher& m) {
+            const auto& root = m.get_match_root();
+            const auto& output = root->output(0);
+            auto inputs = output.get_target_inputs();
+
+            // Check if an output of matched root is consumed as input labeled with reverse_input_channel_index
+            std::vector<ric_attr::Attribute> attrs;
+            for (const auto& input : inputs) {
+                if (ric_attr::has(input)) {
+                    attrs.push_back(ric_attr::get(input).propagate());
+                } else {
+                    return false;
+                }
+            }
+
+            if (attrs.empty())
+                return false;
+
+            // Check that all RIC attrs from consumers can be merged and then merge them
+            auto ric = attrs[0];
+            for (const auto& item : attrs) {
+                if (ric.can_be_merged_with(item)) {
+                    ric.merge_with(item);
+                } else {
+                    return false;
+                }
+            }
+
+            auto data_rank = root->get_output_partial_shape(0).rank().get_length();
+            for (const auto& input : root->inputs()) {
+                auto output = input.get_source_output();
+                const auto& shape = output.get_partial_shape();
+                if (shape.rank().is_dynamic())
+                    return false;
+                const int64_t& shape_rank = shape.rank().get_length();
+                if (shape_rank > data_rank) {
+                    // TODO: handle case when constant input broadcast another one
+                    return false;
+                }
+
+                if (data_rank - shape_rank > ric.get_axis()) {
+                    // we don't have to insert RIC for constant, so we keep propagating
+                    continue;
+                }
+
+                const int64_t& new_axis = ric.get_axis() - (data_rank - shape_rank);
+                const auto& axis_dim = shape[new_axis];
+                if (axis_dim.is_dynamic())
+                    return false;
+                if (axis_dim == 1) {
+                    // we don't have to insert RIC for constant, so we keep propagating
```

Comment:
The comment seems misleading. We don't need RIC here not because of constant, but because channels are 1.

Reply:
changed